kore/src/Kore/Step/RewriteStep.hs

```diff
@@ -284,13 +284,12 @@ applyWithFinalizer
     => Rule.UnifyingRuleVariable rule ~ RewritingVariableName
     => From rule SourceLocation
     => Finalizer rule simplifier
-    -> UnificationProcedure simplifier
     -> [rule]
     -- ^ Rewrite rules
     -> Pattern RewritingVariableName
     -- ^ Configuration being rewritten
     -> simplifier (Results rule)
-applyWithFinalizer finalize unificationProcedure rules initial = do
+applyWithFinalizer finalize rules initial = do
     results <- unifyRules unificationProcedure initial rules
```

Comment:
We can go another step and remove the `UnificationProcedure` argument from `unifyRule`.